lib/export.js

```diff
@@ -753,11 +778,11 @@ function makeRef(id, enclosingNamespace, baseSchemaURL) {
 function makeShrRefObject(refs, baseTypeURL, target = {}) {
   target.type = 'object';
   target.properties = {
-    ShrId: { type: 'string' },
-    EntryId: { type: 'string' },
-    EntryType: { type: 'string' }
+    _ShrId: { type: 'string' },
+    _EntryId: { type: 'string' },
+    _EntryType: { type: 'string' }
```

Comment:
What is the reason for making these prefixed w/ `_`?  I think the entire ref object is something the JSON schema defines itself (not defined by SHR spec) -- so can't we treat its properties like normal?

Reply:
Or is this to indicate the difference between a fully spelled out ShrId (object) vs just the string?

Reply:
Yes, this is to indicate the difference between a full object and a reference. This change was driven by the es6-class-exporter's deserialization implementation. The relevant code there is `json-helper.createInstance()` which was deciding whether or not something is a reference based upon the existence of the `ShrId`, `EntryId`, and `EntryType` fields (see: https://github.com/standardhealth/shr-es6-export/blob/c0c62bfd38b9d8b813b6cb19877113eb09f60020/lib/includes/json-helper.js#L141
). Nuking the namespaces left it assuming that everything was a reference. So prefixing the fields `_` was a quick fix to get things working (see companion PR standardhealth/shr-es6-export#18). I'd be happy to do a more elegant fix, especially since we were considering prefixing all special fields with `_`s as discussed on Slack.